mso/resource_mso_schema_site_anp_epg_domain.go

```diff
@@ -309,36 +362,71 @@ func resourceMSOSchemaSiteAnpEpgDomainCreate(d *schema.ResourceData, m interface
 	anpName := d.Get("anp_name").(string)
 	epgName := d.Get("epg_name").(string)
 	domainType := d.Get("domain_type").(string)
-	domainName := d.Get("dn").(string)
+	vmmDomainType := d.Get("vmm_domain_type").(string)
+	domainNameDnOld := d.Get("dn").(string)
 	deployImmediacy := d.Get("deploy_immediacy").(string)
 	resolutionImmediacy := d.Get("resolution_immediacy").(string)
 
-	var DN, microSegVlanType, portEncapVlanType, vlanEncapMode, switchingMode, switchType, enhancedLagpolicyName, enhancedLagpolicyDn string
+	var DN, microSegVlanType, portEncapVlanType, vlanEncapMode, switchingMode, switchType, enhancedLagpolicyName, enhancedLagpolicyDn, domainName string
 	var microSegVlan, portEncapVlan float64
-	var allowMicroSegmentation bool
+	var allowMicroSegmentation, checkDomainTypeFromDN bool
+
+	_, ok_oldName := d.GetOk("dn")
+	tempVarName, ok_name := d.GetOk("domain_name")
+	tempVarDn, ok_dn := d.GetOk("domain_dn")
+
+	if !ok_oldName && !ok_name && !ok_dn {
+		return fmt.Errorf("domain_dn or domain_name in association with domain_type and vmm_domain_type if applicable are required.")
+	}
 
-	if domainType == "vmmDomain" {
-		DN = fmt.Sprintf("uni/vmmp-VMware/dom-%s", domainName)
+	if ok_name {
+		domainName = tempVarName.(string)
+	} else {
+		domainName = domainNameDnOld
+	}
 
-	} else if domainType == "l3ExtDomain" {
-		DN = fmt.Sprintf("uni/l3dom-%s", domainName)
+	if ok_dn {
+		DN = tempVarDn.(string)
+		vmmp_match, _ := regexp.MatchString("uni/vmmp-.*", DN)
+		checkDomainTypeFromDN = vmmp_match
+		l3dom_match, _ := regexp.MatchString("uni/l3dom-.*", DN)
+		l2dom_match, _ := regexp.MatchString("uni/l2dom-.*", DN)
+		phys_match, _ := regexp.MatchString("uni/phys-.*", DN)
+		fc_match, _ := regexp.MatchString("uni/fc-.*", DN)
```

Comment:
This can be simplified by an expressing like 208-209

Reply:
ok